### PR TITLE
App Bug update 2.0.1

### LINKED ISSUE #207: Scrollbar not visible in Darkmode on some Android phones
Known to be an issue on Android 8.1.0.

https://stackoverflow.com/questions/50831008/control-webview-scrollbar-position-color

### LINKED ISSUE #240: white background when rotating to landscape in darkmode
Small regression

reproduce:
enable darkmode
rotate phone to lamdscape

to see it even better, use ios simulator and set animation speed to slow.

### LINKED ISSUE #241: Support Hermes on Android
Update RN to latest and enable hermes render engine on android.

benefits: "improved start-up time, decreased memory usage, and smaller app size"

https://reactnative.dev/docs/hermes

### LINKED ISSUE #246: Crash related to allowFullscreenVideo on Android Webview
Error: https://play.google.com/console/u/0/developers/5027121202214205556/app/4975641939341220182/vitals/crashes/9dad1afa/details?installedFrom=PLAY_STORE&days=30

java.lang.NullPointerException
com.reactnativecommunity.webview.RNCWebViewManager.setupWebChromeClient

patch fix: https://github.com/react-native-webview/react-native-webview/issues/1406

maybe fixed by: allowsFullscreenVideo:  true

### LINKED ISSUE #247: Crashes related to JSC Engine
https://play.google.com/console/u/0/developers/5027121202214205556/app/4975641939341220182/vitals/crashes/9dad1afa/details?installedFrom=PLAY_STORE&days=30

https://play.google.com/console/u/1/developers/5027121202214205556/app/4975641939341220182/vitals/crashes/c023f221/details?installedFrom=PLAY_STORE&days=30&versionCode=135

https://bugs.webkit.org/show_bug.cgi?id=200983
Maybe fixed by using hermes:
2) Use RN 0.60.x with Hermes: The current version of RN 0.60.x comes with its own stability issues and until those get resolved, we cannot move to RN 0.60.

### LINKED ISSUE #248: trackplayer crash
https://play.google.com/console/u/1/developers/5027121202214205556/app/4975641939341220182/vitals/crashes/7c483523/details?installedFrom=PLAY_STORE&days=30&versionCode=135

https://github.com/react-native-kit/react-native-track-player/issues/391
https://github.com/react-native-kit/react-native-track-player/pull/903

probable fix:
stopWithApp: true

### LINKED ISSUE #249: Android App doesn't work in Chrome-OS
User Reporst that the android app doesn't work in chrome OS

Possible solution:
https://stackoverflow.com/questions/59175609/how-to-make-android-app-available-on-chromebook-react-native

### LINKED ISSUE #250: Scrollbar is invisible on Android (non-darkmode)
Scrollbar is not visible in bright-mode.

### PR DESCRIPTION
**Pull request by ovbm**
fixes #249, fixes #250, fixes #207, fixes #248, fixes #247, fixes #246, fixes #241, fixes #240

### CODE REVIEW

**Comment by ovbm** [comment]
@tpreusse 
A lot of the file changes are due to the react-native upgrade. I was able to build and run the app on both ios simulator and android simulators. Would be good to make a test flight version.
I'm currently blocked because of a weird node-related bug, which prevents me from building the ios bundle with the command line (only works via xcode, see: https://github.com/facebook/react-native/issues/31259 for details). Could you pull and build on your end?

File: ios/republikapp/AppDelegate.m [hunk 1]
```diff
@@ -30,16 +30,19 @@ @implementation AppDelegate
...
   RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge
                                                    moduleName:@"republikapp"
                                             initialProperties:nil];
-
-  rootView.backgroundColor = [[UIColor alloc] initWithRed:0.0f green:0.0f blue:0.0f alpha:1];
+  if (@available(iOS 13.0, *)) {
+    rootView.backgroundColor = [UIColor blackColor];
+  } else {
+    rootView.backgroundColor = [UIColor blackColor];
+  }
```
Comment (tpreusse) [inline ios/republikapp/AppDelegate.m:45]: Is the if statement really necessary?

```suggestion
  rootView.backgroundColor = [UIColor blackColor];
```

**Review by tpreusse — APPROVED** [review]
Code LGTM.

Just for sanity a few question re Android testing:
- Did you test linking opening on Android extensively? Including when app is shut down and also open in the background?
- Also can video still play go into full screen? E.g. An der Bar? Can the gallery still go full screen?

And out of curiosity: can you post a screenshot of the grey scrollbar on Android?

Thank you and general approval to ship from my side.

**Comment by ovbm** [comment]
Tested fullscreen video and gallery. Working as intended on my device.

ScrollBars:

Screenshot_20210409-162709_Republik
Screenshot_20210409-162649_Republik

File: ios/republikapp/AppDelegate.m (hunk 1 above)
Comment (ovbm) [reply ios/republikapp/AppDelegate.m:45]: duh 😂

**Comment by ovbm** [comment]
Linking needs to be tested more, agreed

**Comment by ovbm** [comment]
Deep linking works perfectly :), will merge and release as there hasn't been any reported issues.

**Pull request merged**